src/dclass.d

```diff
@@ -922,12 +922,15 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
         //    this() { }
         if (!ctor && baseClass && baseClass.ctor)
         {
-            auto fd = resolveFuncCall(loc, sc2, baseClass.ctor, null, null, null, 1);
+            auto fd = resolveFuncCall(loc, sc2, baseClass.ctor, null, type, null, 1);
+            if (!fd) // try shared base ctor instead
+                fd = resolveFuncCall(loc, sc2, baseClass.ctor, null, type.sharedOf, null, 1);
```

Comment:
Handling shared base ctor was added to not break existing code, even though it never worked to actually instantiate a class like Derived3 (b/c of the unshared implicit ctor).
Implicitly declaring overloaded ctors is out of scope for the bugfix and is a topic for [Issue 9066 – Add constructor inheritance feature](https://issues.dlang.org/show_bug.cgi?id=9066).